control/server.py

```diff
@@ -163,18 +164,20 @@ def _start_spdk(self):
             cmd += shlex.split(spdk_tgt_cmd_extra_args)
         self.logger.info(f"Starting {' '.join(cmd)}")
         try:
-            self.spdk_process = subprocess.Popen(cmd,
-                                                 preexec_fn=set_pdeathsig(
-                                                     signal.SIGTERM))
+            # install SIGCHLD handler
+            signal.signal(signal.SIGCHLD, sigchld_handler)
+
+            # start spdk process
+            self.spdk_process = subprocess.Popen(cmd)
         except Exception as ex:
             self.logger.error(f"Unable to start SPDK: \n {ex}")
             raise
 
         # Initialization
         timeout = self.config.getfloat("spdk", "timeout")
         log_level = self.config.get("spdk", "log_level")
-        conn_retries = self.config.getint_with_default("spdk",
-                                                       "conn_retries", 10)
+        # connect timeout: spdk client reties 5 times per sec
+        conn_retries = int(timeout * 5)
         self.logger.info({
             f"Attempting to initialize SPDK: rpc_socket: {spdk_rpc_socket},",
             f" conn_retries: {conn_retries}, timeout: {timeout}",
```

Comment:
BTW, this message looks like below:
```
INFO:control.server:{'Attempting to initialize SPDK: rpc_socket: /var/tmp/spdk.sock,', ' conn_retries: 300, timeout: 60.0'}
```
I think the curly brackets around the formatted message string is wrong here.